kratos/geometries/quadrilateral_3d_4.h

```diff
@@ -608,15 +599,15 @@ template<class TPointType> class Quadrilateral3D4
 
             rResult[0] += DeltaXi[0];
             rResult[1] += DeltaXi[1];
-            rResult[2] = 0.0;
 
             if ( norm_2( DeltaXi ) > 300 )
             {
                 res[0] = 0.0;
                 res[1] = 0.0;
+            #ifdef KRATOS_DEBUG 
                 std::cout << "detJ =" << det_j << "DeltaX = " << DeltaXi << " stopping calculation and assigning the baricenter" << std::endl;
+            #endif  
                 break;
-                //KRATOS_ERROR << "Computation of local coordinates failed at iteration" << k << std:endl;

```

Comment:
This is changing the behavior and I'm not sure if is a good idea to return a zero with just a small hint....

Reply:
Ok